src/main/java/io/vertx/core/impl/TaskQueue.java

```diff
@@ -125,19 +125,30 @@ public Consumer<Runnable> unschedule() {
    */
   public void execute(Runnable task, Executor executor) {
     synchronized (tasks) {
-      tasks.add(new ExecuteTask(task, executor));
+      ExecuteTask executeTask = new ExecuteTask(task, executor);
+      tasks.add(executeTask);
```

Comment:
I think instead since we are in a synchronized block, we could move this addition to the bottom of the block after we are sure the executor accepted the task instead of adding/removing it